docs/designs/login-design.md

```diff
@@ -0,0 +1,165 @@
+- A mechanism to retrieve forgotten PINs via email.
+- A database schema supporting volunteer sign-ins.
+
+The functionalities of the admin dashboard, volunteer dashboard, setting user activation status, and enrolling or deleting users are outside the scope of this document.
+
+### 1.3 Definitions, Acronyms, and Abbreviations
+- **OAuth**: Open Authorization, a protocol for secure user authentication.
+- **PIN**: Personal Identification Number.
+
+### 1.4 References
+- Azure OAuth Documentation
+- User Interface Design Guidelines
+
+## 2. System Overview
+The system authenticates users via Azure OAuth, enforcing that the email domain is @plymouthhousing.org. After authentication:
+- Volunteers are redirected to a PIN page (such as volunteers@plymouthhousing.org).
+- Administrators are redirected to the admin dashboard (for accounts not using the volunteers@plymouthhousing.org domain).
+
+Volunteers will select their name from a dropdown and enter a 4-digit PIN to sign in.
+
+## 3. Architecture Goals and Constraints
+
+### 3.1 Goals
+- Secure sign-in via Azure OAuth.
+- Role-based redirection.
+- Simple PIN-based authentication for volunteers.
+- Data integrity and security for volunteer information.
+- Mechanism for PIN recovery via email.
+
+
+### 3.2 Constraints
+- Must use Azure OAuth for authentication.
+- Admin and volunteer dashboards are outside the scope of this design.
+- The solution must comply with data protection and privacy regulations.
+- The process for resetting a volunteer's PIN has not been finalized. It could be managed by admins via the admin page or by the volunteers themselves through a new PIN reset page.
+
+## 4. Architectural Representations
+
+### 4.1 High-Level Architecture Diagram
+The high-level architecture includes the following components:
+- **Sign-In Page**: Redirects to Azure OAuth.
+- **Azure OAuth Service**: Authenticates users based on their email.
+- **Volunteer Name Page**: Allows volunteers to select name using a dropdown.
+- **Volunteer PIN Page**: Allows volunteers to sign in using a PIN.
+- **Email Notification**: Sends email notifications for user to reset the PIN.
+- **Database**: Stores volunteer details, including the PIN, activation status, and last sign-in date.
+
+### 4.2 Component Descriptions
+- **Sign-In Page**: Initial landing page for all users; redirects to Azure OAuth for authentication.
+- **Azure OAuth Service**: Handles secure authentication; determines user role based on the email domain.
+- **Volunteer Name Page**: Displays a dropdown of volunteers names.
+- **Volunteer PIN Page**: Displays a PIN field; checks entered data against the database.
+- **Email Notification via Resend API**:  Sends email notifications using the Resend API through Supabase Edge Functions. This feature is used for sending important notifications, such as PIN reset emails to volunteers. [Resend API Documentation](https://resend.com/docs/send-with-supabase-edge-functions)
```

Comment:
Is this pending a discussion where PH wants to host? If they host on-prem, do we want/need another solution?

Reply:
I agree with you. It is important to figure out what they want.